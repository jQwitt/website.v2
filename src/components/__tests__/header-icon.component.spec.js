import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { HeaderIcon } from '../index';

afterEach(cleanup);
describe('HeaderIcon renders correctly', () => {
  test('snapshot', () => {
    const { asFragment } = render(<HeaderIcon />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('conditionally renders icon at any size', () => {
    const data = {
      icon: 'subTract',
      size: ['xsmall', 'small', 'medium', 'large', 'xlarge', 'jumbo'],
    };
    data.size.forEach((cl) => {
      const { container } = render(<HeaderIcon icon={data.icon} size={cl} />);
      expect(container.querySelector('i').classList.contains(cl)).toBe(true);
    });
  });

  it('conditionally renders underline', () => {
    const data = {
      underline: true,
    };
    const { container } = render(<HeaderIcon underline={data.underline} />);
    expect(container.firstChild.classList.contains('underlined')).toBe(true);
  });
});
