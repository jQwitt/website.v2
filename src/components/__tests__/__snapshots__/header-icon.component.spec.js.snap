// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderIcon renders correctly snapshot 1`] = `
<DocumentFragment>
  <div
    class="ml-3 col-6 header-icon"
  >
    <i
      class="flaticon-add "
    />
  </div>
</DocumentFragment>
`;
